src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,688 @@
+     *
+     *   @return Success if the allocation is successful and a PushTransportConnectionID was produced; otherwise, the command SHALL
+     * be rejected with Failure.
+     *
+     *   The buffers storing URL, Trigger Options, Motion Zones, Container Options is owned by the PushAVStreamTransport Server.
+     *   The buffers are allocated on success of AllocatePushTransport and deallocated on success of DeallocatePushTransport
+     * command. The delegate is expected to process the transport following transport options: URL :  Validate the URL
+     *   StreamUsage,VideoStreamID,AudioStreamID for selection of Stream.
+     *   Allocate the transport and map it to the connectionID.
+     *   On Success TransportConfigurationStruct is sent as response by the server.
+     */
+    virtual Protocols::InteractionModel::Status AllocatePushTransport(const TransportOptionsStruct & transportOptions,
+                                                                      const uint16_t connectionID) = 0;
+    /**
+     *   @brief Handle Command Delegate for Stream transport deallocation for the
+     *   provided connectionID.
+     *
+     *   @param connectionID[in]        Indicates the connectionID to deallocate.
+     *
+     *   @return Success if the transport deallocation is successful; otherwise, the delegate is expected to return status code BUSY
+     *   if the transport is currently uploading.
+     *
+     */
+    virtual Protocols::InteractionModel::Status DeallocatePushTransport(const uint16_t connectionID) = 0;
+    /**
+     *   @brief Handle Command Delegate for Stream transport modification.
+     *
+     *   @param connectionID [in]           Indicates the connectionID of the stream transport to modify.
+     *
+     *   @param transportOptions [out]      represents the Transport Options to modify.
+     *
+     *   The buffers storing URL, Trigger Options, Motion Zones, Container Options is owned by the PushAVStreamTransport Server.
+     *   The allocated buffers are cleared and reassigned on success of ModifyPushTransport and deallocated on success of
+     * DeallocatePushTransport.
+     *
+     *   @return Success if the stream transport modification is successful; otherwise, the command SHALL be rejected with Failure.
+     */
+    virtual Protocols::InteractionModel::Status
+    ModifyPushTransport(const uint16_t connectionID, const Structs::TransportOptionsStruct::DecodableType transportOptions) = 0;
+
+    /**
+     *   @brief Handle Command Delegate for Stream transport modification.
+     *
+     *   @param connectionIDList [in]       represent the list of connectionIDs for which new transport status to apply.
+     *   @param transportStatus  [in]       represents the updated status of the connection(s).
+     *
+     *  If the transportStatus field is set to Inactive(1).Disable transmissions on the transport.Remove any queued items for
+     *  upload.Cancel any uploads currently in progress.
```

Comment:
This seems to be formatted really weirdly...  And some of the bits below too.  Missing spaces after end of sentence?